src/univariate/discrete/binomial.jl

```diff
@@ -67,11 +67,36 @@ params(d::Binomial) = (d.n, d.p)
 
 mean(d::Binomial) = ntrials(d) * succprob(d)
 var(d::Binomial) = ntrials(d) * succprob(d) * failprob(d)
-function mode(d::Binomial{T}) where T<:Real
+
+function mode(d::Binomial)
+    (n, p) = params(d)
+    v = (n + 1) * p
+    quasi_mode = floor(Int, v)
+    if quasi_mode == v
+        if p == 1
+            n
+        else
+            quasi_mode-1
+        end
+    else
+        quasi_mode
+    end
```

Comment:
AFAICT alternatives with fewer branches would be
```suggestion
    iszero(p) ? 0 : ceil(Int, (n + 1) * p) - 1
```
or
```suggestion
    max(0, ceil(Int, (n + 1) * p) - 1)
```
or
```suggestion
    clamp(ceil(Int, (n + 1) * p) - 1, 0, n)
```

Reply:
There is little to no difference in runtime between these options (or the option in the PR).  The main difference is in the max runtime, and the PR options and the `max(0, ceil(...))` have the shorter max runtime (the two are nearly identical with a max of about 10x the average runtime), while the max runtime for the other two options is about twice as long (about 20x the average runtime).  The standard deviations show a similar pattern (but they are smaller than the average runtime).

Given there isn't much of a different in run times, I propose the PR version is preferable because it is easiest to read and reason about.

Reply:
@devmotion Is there another reason to consider fewer branches other than runtime?